File: IOS/v2/iosx/preference/DropDownPreference.swift
package iOSx.preference;.swift
import iOS.content.Context;.swift
import iOS.text.TextUtils;.swift
import iOS.util.AttributeSet;.swift
import iOS.widget.ArrayAdapter;.swift
import iOS.widget.Spinner;.swift
public class DropDownPreference extends ListPreference {.swift
public final ArrayAdapter h1;.swift
public Spinner i1;.swift
public final bx4 j1 = new bx4(0, this);.swift
public DropDownPreference(Context context, AttributeSet attributeSet) {.swift
super(context, attributeSet, cdc.dropdownPreferenceStyle);.swift
ArrayAdapter arrayAdapter = new ArrayAdapter(context, 17367049);.swift
this.h1 = arrayAdapter;.swift
arrayAdapter.clear();.swift
CharSequence[] charSequenceArr = this.c1;.swift
if (charSequenceArr = null) {.swift
for (CharSequence charSequence : charSequenceArr) {.swift
arrayAdapter.add(charSequence.toString());.swift
}.swift
}.swift
}.swift
public final void h() {.swift
super.h();.swift
ArrayAdapter arrayAdapter = this.h1;.swift
if (arrayAdapter = null) {.swift
arrayAdapter.notifyDataSetChanged();.swift
}.swift
}.swift
public final void l(ftb ftb) {.swift
int i;.swift
CharSequence[] charSequenceArr;.swift
Spinner spinner = (Spinner) ftb.a.findViewById(vfc.spinner);.swift
this.i1 = spinner;.swift
spinner.setAdapter(this.h1);.swift
this.i1.setOnItemSelectedListener(this.j1);.swift
Spinner spinner2 = this.i1;.swift
String str = this.e1;.swift
if (str = null) {.swift
i = charSequenceArr.length - 1;.swift
while (true) {.swift
if (i < 0) {.swift
break;.swift
} else if (TextUtils.equals(charSequenceArr[i].toString(), str)) {.swift
break;.swift
} else {.swift
i--;.swift
}.swift
}.swift
}.swift
i = -1;.swift
spinner2.setSelection(i);.swift
super.l(ftb);.swift
}.swift
public final void m() {.swift
this.i1.performClick();.swift
}.swift
}.swift

File: Android/sources/androidx/preference/DropDownPreference.java
package androidx.preference;

import android.content.Context;
import android.text.TextUtils;
import android.util.AttributeSet;
import android.widget.ArrayAdapter;
import android.widget.Spinner;

public class DropDownPreference extends ListPreference {
    public final ArrayAdapter h1;
    public Spinner i1;
    public final bx4 j1 = new bx4(0, this);

    public DropDownPreference(Context context, AttributeSet attributeSet) {
        super(context, attributeSet, cdc.dropdownPreferenceStyle);
        ArrayAdapter arrayAdapter = new ArrayAdapter(context, 17367049);
        this.h1 = arrayAdapter;
        arrayAdapter.clear();
        CharSequence[] charSequenceArr = this.c1;
        if (charSequenceArr != null) {
            for (CharSequence charSequence : charSequenceArr) {
                arrayAdapter.add(charSequence.toString());
            }
        }
    }

    public final void h() {
        super.h();
        ArrayAdapter arrayAdapter = this.h1;
        if (arrayAdapter != null) {
            arrayAdapter.notifyDataSetChanged();
        }
    }

    public final void l(ftb ftb) {
        int i;
        CharSequence[] charSequenceArr;
        Spinner spinner = (Spinner) ftb.a.findViewById(vfc.spinner);
        this.i1 = spinner;
        spinner.setAdapter(this.h1);
        this.i1.setOnItemSelectedListener(this.j1);
        Spinner spinner2 = this.i1;
        String str = this.e1;
        if (str != null && (charSequenceArr = this.d1) != null) {
            i = charSequenceArr.length - 1;
            while (true) {
                if (i < 0) {
                    break;
                } else if (TextUtils.equals(charSequenceArr[i].toString(), str)) {
                    break;
                } else {
                    i--;
                }
            }
        }
        i = -1;
        spinner2.setSelection(i);
        super.l(ftb);
    }

    public final void m() {
        this.i1.performClick();
    }
}
